pywbem_mock/_baseprovider.py

```diff
@@ -333,13 +295,114 @@ def remove_namespace(self, namespace):
                 CIM_ERR_NAMESPACE_NOT_EMPTY,
                 _format("Namespace {0!A} contains objects.", namespace))
 
+    @staticmethod
+    def interop_namespace_names():  # pylint: disable=no-self-use
+        """
+        Returns an iterable of the valid interop namespace names where
+        the names are considered case insensitive.
+
+        Returns:
+
+          List of :term:`string` containing the valid names for
+          namespaces that can be interop namespaces. Any name not on this list
+          cannot be an interop provider
+
+        """
+        return WBEMServer.INTEROP_NAMESPACES
+
+    def is_interop_namespace(self, namespace):  # pylint: disable=no-self-use
+        """
+        Tests if the namespace parameter defines a namespace name that is one
+        of the allowed set of DMTF names for the interop namespace.
+
+        This test is independent of the test in WBEMServer since it is just
+        confirming that there is an namespace with one of the valid interop
+        namespace names in the physical cim repository.
+
+        Parameters:
+
+            namespace (:term:`string`):
+                The namespace name that is to be tested.
+
+        Returns:
+            True: If the name is one of the valid interop namespace names
+            False: if the name is not an interop namespace valid name.
+
+        """
+        ns_lower = [ns.lower() for ns in self.interop_namespace_names()]
+        return namespace.lower() in ns_lower
+
+    def find_interop_namespace(self):
+        """
+        Find  an interop namespace if one exists in the CIM repository.
+        The interop namespace is identified by it name and the list of
+        possible names is defined by the method `interop_namespace_names`.
+
+        Returns:
+          The name of the interop namespace if one exists or None if there
+          is no interop namespace in the CIM repository.
```

Comment:
Add the return type string to make Sphinx recognize it properly:
```
        Returns:
          :term:`string`: The name of the ...
```

Reply:
DONE